### DISCUSSION TITLE
FATAL: Ansible CLI (2.10.2) and python module (2.10.6) versions do not match.

### DISCUSSION THREAD

**Discussion by west1nat0r** [ansible/ansible-lint#1468] [Q&A] [answered]
Hi togehter,

I would love to get Ansible-Lint running on my environment, but it does not execute. When trying to execute `ansible-lint` in my command line it throws me the error: 

`FATAL: Ansible CLI (2.10.2) and python module (2.10.6) versions do not match.`

I kindly checked out the discussions and already digged deeper into the code to see where the FATAL Error comes from, but I am not capable to get it running. 

Just before you ask: 
`Ansible --version` : ansible 2.10.2 , python version = 3.8.7 (default, Feb 16 2021, 10:40:48) [Clang 11.0.3 (clang-1103.0.32.62)]

I already upgraded ansible and ansible-lint through pip and I honestly do not know how to downgrade the ansible python module. I also tried to install ansible v. 2.10.2, but it didn't work as well.

Thanks in advance for your support!

**Comment by west1nat0r** [comment, marked as answer]
I finally found the problem. I had several python/bin/ directories and somehow Ansible CLI was loaded from another bin than the ansible module!

For all people having the same problem:
`find / -name ansible 2>/dev/null`
and check every executable with `--version` flag. Finally edit your `$PATH` variable to match.

**Comment by ericcames** [comment]
Hey folks,

I am trying to get ansible-lint to work on my RHEL 8 Laptop.  I am not sure what I am doing wrong.  Here is what I am seeing:

find / -executable -name ansible -type f 2>/dev/null -exec '{}' --version \; | grep '^ansible \[core\|executable'
ansible [core 2.11.12] 
  executable location = /home/eames/.local/share/containers/storage/overlay/d5fc36a694a6f2c01d90dcbb481f209aaa4ef0961f8b516c818d6e98a6e0048d/diff/usr/local/bin/ansible
ansible [core 2.12.2]
  executable location = /home/eames/.local/share/containers/storage/overlay/a5b2fdc87e6ef562f64078edeb6daa8b8a279beac12b430e16c8501ea49b3d38/diff/usr/bin/ansible
ansible [core 2.12.2]
  executable location = /home/eames/.local/share/containers/storage/overlay/b49ac1913b74bd906130d13fa1125b254f118771324d7b085623ab6471702b3c/diff/usr/bin/ansible
ansible [core 2.11.12] 
  executable location = /home/eames/.local/bin/ansible
ansible [core 2.12.2]
  executable location = /usr/bin/ansible

ansible-lint --version
WARNING: PATH altered to include /usr/bin
FATAL: Ansible CLI (2.12.2) and python module (2.11.12) versions do not match. This indicates a broken execution environment.

grep 'PATH=$PATH' .bashrc
PATH=$PATH:/home/eames/.local/share/containers/storage/overlay/d5fc36a694a6f2c01d90dcbb481f209aaa4ef0961f8b516c818d6e98a6e0048d/diff/usr/local/bin:/home/eames/.local/share/containers/storage/overlay/a5b2fdc87e6ef562f64078edeb6daa8b8a279beac12b430e16c8501ea49b3d38/diff/usr/bin:/home/eames/.local/share/containers/storage/overlay/b49ac1913b74bd906130d13fa1125b254f118771324d7b085623ab6471702b3c/diff/usr/bin:/home/eames/.local/bin:/usr/bin


Thanks in advance for your feedback!!

**Comment by kema-dev** [comment]
For anyone using linuxbrew, I solved this with `brew reinstall ansible` that warned about a broken link, fixed with `brew link --overwrite ansible` !
